packages/appkit/src/utils/ConfigUtil.ts

```diff
@@ -0,0 +1,218 @@
+import type { SocialProvider } from '@reown/appkit-common'
+import { AlertController, ApiController, ConstantsUtil } from '@reown/appkit-controllers'
+import type { FeatureID, RemoteFeatures, TypedFeatureConfig } from '@reown/appkit-controllers'
+
+import type { AppKitOptionsWithSdk } from '../client/appkit-base-client.js'
+
+export const ConfigUtil = {
+  async fetchRemoteFeatures(config: AppKitOptionsWithSdk): Promise<RemoteFeatures> {
+    const localFeatures = config.features || {}
+    const { DEFAULT_REMOTE_FEATURES } = ConstantsUtil
+    let remoteFeaturesConfig: RemoteFeatures = { ...DEFAULT_REMOTE_FEATURES }
+
+    let apiProjectConfig: TypedFeatureConfig[] | null = null
+    let isApiCallSuccessful = false
+    const localSettingsOverridden = new Set<string>()
+
+    function getApiConfig<T extends FeatureID>(id: T) {
+      if (!apiProjectConfig) {
+        return undefined
+      }
+
+      return apiProjectConfig.find((f): f is Extract<TypedFeatureConfig, { id: T }> => f.id === id)
```

Comment:
is this not returning the project config already? why do we need to filter by id?